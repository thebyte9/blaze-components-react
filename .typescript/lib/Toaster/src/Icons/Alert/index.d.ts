/// <reference types="react" />
declare const AlertIcon: (props: any) => JSX.Element;
export default AlertIcon;
