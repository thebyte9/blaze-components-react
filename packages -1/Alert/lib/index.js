"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
exports.__esModule = true;
var button_1 = __importDefault(require("@blaze-react/button"));
var utils_1 = __importDefault(require("@blaze-react/utils"));
var react_1 = __importStar(require("react"));
var Alert = function (_a) {
    var _b;
    var children = _a.children, close = _a.close, icon = _a.icon, type = _a.type, classNames = _a.utils.classNames, attrs = __rest(_a, ["children", "close", "icon", "type", "utils"]);
    var _c = react_1.useState(false), offModal = _c[0], setModalOff = _c[1];
    var alertClassName = classNames("alert", (_b = {
            "alert--dismissable": close,
            "alert--icon": !!icon
        },
        _b["alert--" + type] = !!type,
        _b));
    var renderAlert = (react_1["default"].createElement("div", __assign({ className: alertClassName }, attrs),
        icon && react_1["default"].createElement("i", { className: "material-icons" }, icon),
        children,
        close && (react_1["default"].createElement(button_1["default"], { onClick: function () { return setModalOff(true); }, className: "icon-button icon-button--close" },
            react_1["default"].createElement("i", { className: "material-icons" }, "close")))));
    return react_1["default"].createElement(react_1.Fragment, null, !offModal && renderAlert);
};
Alert.defaultProps = {
    children: "No content",
    close: false,
    icon: "",
    type: ""
};
exports["default"] = utils_1["default"](Alert);
