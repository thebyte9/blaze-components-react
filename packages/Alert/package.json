{
  "name": "@blaze-react/alert",
  "version": "0.4.11",
  "publishConfig": {
    "access": "public"
  },
  "description": "",
  "main": "./lib/index.js",
  "scripts": {
    "build": "../../node_modules/.bin/tsc --jsx react --esModuleInterop ./src/* --outDir lib ",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "../../node_modules/.bin/tslint -c ../../tslint.json src/**/*.tsx",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "pre-publish": "npm run build && npm run lint && npm run test"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@blaze-react/button": "^0.4.11",
    "@blaze-react/utils": "^0.4.11"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.5",
    "@blaze-react/blaze-components-theme": "^0.4.11"
  },
  "peerDependencies": {
    "prop-types": "^15.7.2",
    "react": "^16.9.0",
    "react-dom": "^16.9.0"
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      "@babel/preset-react"
    ]
  },
  "gitHead": "986a601913a963e035b116f643ed1d9230ea1a87"
}
