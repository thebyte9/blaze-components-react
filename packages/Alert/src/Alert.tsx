import { ButtonView } from '@blaze-react/button';
import React, { Fragment, FunctionComponent, useState } from 'react';
import { buildClassNames } from '@blaze-react/utils';

interface IAlertProps {
  icon?: string;
  type?: string;
  close?: boolean;
  children?: JSX.Element | string;
}

const Alert: FunctionComponent<IAlertProps> = ({ children, close, icon, type, ...attrs }): JSX.Element => {
  const [offModal, setModalOff] = useState<boolean>(false);

  const alertClassName: string = buildClassNames('alert', {
    'alert--dismissable': close,
    'alert--icon': !!icon,
    [`alert--${type}`]: !!type,
  });

  const renderAlert = (
    <div className={alertClassName} {...attrs}>
      {icon && <i className="material-icons">{icon}</i>}
      {children}
      {close && (
        <ButtonView onClick={() => setModalOff(true)} className="icon-button icon-button--close">
          <i className="material-icons">close</i>
        </ButtonView>
      )}
    </div>
  );
  return <Fragment>{!offModal && renderAlert}</Fragment>;
};

Alert.defaultProps = {
  children: 'No content',
  close: false,
  icon: '',
  type: '',
};

export default Alert;
