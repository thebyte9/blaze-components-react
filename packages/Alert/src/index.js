import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import Button from '@blaze-react/button';

const Alert = ({ children, close, icon, type, ...attrs }) => {
  const [offModal, setModalOff] = useState(false);
  const assignType = type && `alert--${type}`;
  const isDismissable = close && 'alert--dismissable';
  const withIcon = icon && 'alert--icon';

  const renderAlert = (
    <div className={`alert ${assignType} ${isDismissable} ${withIcon}`} {...attrs}>
      {icon && <i className="material-icons">{icon}</i>}
      {children}
      {close && (
        <Button onClick={() => setModalOff(true)} className="icon-button icon-button--close">
          <i className="material-icons">close</i>
        </Button>
      )}
    </div>
  );

  return <Fragment>{!offModal && renderAlert}</Fragment>;
};

Alert.propTypes = {
  icon: PropTypes.string,
  type: PropTypes.string,
  close: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Alert.defaultProps = {
  icon: '',
  type: '',
  close: false,
  children: 'No content'
};

export default Alert;
