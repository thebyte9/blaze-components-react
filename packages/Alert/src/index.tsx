import Button from "@blaze-react/button";
import WithUtils from "@blaze-react/utils";
import React, { Fragment, FunctionComponent, useState } from "react";
interface IAlertProps {
  icon?: string;
  type?: string;
  close?: boolean;
  utils: {
    buildClassNames: (...args: any) => string;
  };
  children?: JSX.Element | string;
}

const Alert: FunctionComponent<IAlertProps> = ({
  children,
  close,
  icon,
  type,
  utils: { buildClassNames },
  ...attrs
}): JSX.Element => {
  const [offModal, setModalOff] = useState<boolean>(false);

  const alertClassName: string = buildClassNames("alert", {
    "alert--dismissable": close,
    "alert--icon": !!icon,
    [`alert--${type}`]: !!type
  });

  const renderAlert = (
    <div className={alertClassName} {...attrs}>
      {icon && <i className="material-icons">{icon}</i>}
      {children}
      {close && (
        <Button
          onClick={() => setModalOff(true)}
          className="icon-button icon-button--close"
        >
          <i className="material-icons">close</i>
        </Button>
      )}
    </div>
  );
  return <Fragment>{!offModal && renderAlert}</Fragment>;
};
Alert.defaultProps = {
  children: "No content",
  close: false,
  icon: "",
  type: ""
};
export default WithUtils(Alert);
