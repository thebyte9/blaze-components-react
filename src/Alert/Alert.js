import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';

const Alert = ({
  children,
  close,
  icon,
  type,
  ...attrs
}) => {
  const [offModal, setModalOff] = useState(false);
  const assignType = type && `alert--${type}`;
  const isDismissable = close && 'alert--dismissable';
  const withIcon = icon && 'alert--icon';

  const renderAlert = (
    <div className={`alert ${assignType} ${isDismissable} ${withIcon}`} {...attrs}>
      {icon && <i className="material-icons">{icon}</i>}
      {children}
      {close && (
      <button onClick={() => setModalOff(true)} className="icon-button icon-button--close" type="button">
        <i className="material-icons">close</i>
      </button>
      )}
    </div>
  );

  return (
    <Fragment>
      {!offModal && renderAlert}
    </Fragment>
  );
};

Alert.propTypes = {
  close: PropTypes.bool,
  icon: PropTypes.string,
  type: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Alert.defaultProps = {
  close: false,
  icon: '',
  type: '',
  children: 'No content'
};

export default Alert;
