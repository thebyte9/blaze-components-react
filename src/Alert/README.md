
## Description

Provide contextual feedback messages for typical user actions with the handful of available and flexible alert messages. Alerts may have a type, be dismissable, include a close button, and contain any sort of children components.

## Usage

* Simple

```js
<Alert>This is a simple alert.</Alert>
```

* With modifier

```js
<Alert type="info">This is a info alert.</Alert>
```
* With icon

```js
<Alert icon="error" 
  type="warning">
  Warning alert with icon
</Alert>
```

* Dismissable

```js
<Alert close 
  type="success">
  This is a success alert that is closable.
</Alert>
```

* With extra content

```js
<Alert type="primary">
  <p>Primary alert with content.</p>
  <ol>
    <li>lorem</li>
    <li>ipsum</li>
  </ol>
</Alert>
```

## API

##### Alert can receive a number of `props` as follow:

| NAME   | TYPE | DEFAULT | OPTIONS |
| :---  | :---:  | :---: | ------- |
| type | string | empty      | primary, secondary, success, warning, dark, info, light|
| icon | string | empty      | Material icons |
| close | Boolean | false      | 
| children | single/array of nodes | 'No content' |
