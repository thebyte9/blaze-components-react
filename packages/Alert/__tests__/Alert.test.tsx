import { fireEvent, render } from "@testing-library/react";
import { shallow } from "enzyme";
import expect from "expect";
import React from "react";
import Alert from "../src";

describe("Alert component", () => {
  test("should be defined and renders correctly (snapshot)", () => {
    const wrapper = shallow(<Alert />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test("should close alert", () => {
    const { container, getByText } = render(
      <Alert close icon="error" type="warning" />
    );

    const close = getByText("close");
    fireEvent.click(close);

    expect(container.childElementCount).toBe(0);
  });
});
