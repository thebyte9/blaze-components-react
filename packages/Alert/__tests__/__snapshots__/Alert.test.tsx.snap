// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should be defined and renders correctly (snapshot) 1`] = `
<Alert
  close={false}
  icon=""
  type=""
  utils={
    Object {
      "ErrorMessage": [Function],
      "buildClassNames": [Function],
      "removeExtraSpaces": [Function],
      "uniqueId": [Function],
    }
  }
>
  No content
</Alert>
`;
